modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb

```diff
@@ -35,21 +35,26 @@ def submit_form_2122 # rubocop:disable Metrics/MethodLength
           file_number = check_file_number_exists!
           claimant_information = validate_dependent_claimant!(poa_code:)
 
-          power_of_attorney = ClaimsApi::PowerOfAttorney.find_using_identifier_and_source(header_md5:,
-                                                                                          source_name:)
+          primary_identifier = {}
+          primary_identifier[:header_hash] = header_hash if header_hash.presence
+          primary_identifier[:header_md5] = header_md5 unless header_hash.presence
```

Comment:
What do you think of:
```suggestion
          primary_identifier =
            if header_hash.present?
              { header_hash: header_hash }
            else
              { header_md5: header_md5 }
            end
```